homeassistant/components/templated_automation/__init__.py

```diff
@@ -0,0 +1,106 @@
+"""
+Support for Templated Automation.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/templated_automation/
+"""
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.binary_sensor import \
+    DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.input_boolean import \
+    DOMAIN as INPUT_BOOLEAN_DOMAIN
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.const import CONF_SOURCE, STATE_OFF, \
+    SERVICE_TURN_OFF, SERVICE_TURN_ON, ATTR_ENTITY_ID
+from homeassistant.core import callback
+from homeassistant.helpers.event import async_track_state_change
+
+CONF_BIDIRECTIONAL = 'bidirectional'
+CONF_INVERTED = 'inverted'
+CONF_TARGET = 'target'
+
+DEFAULT_BIDIRECTIONAL = False
+DEFAULT_INVERTED = False
+
+DOMAIN = 'templated_automation'
+
+BINARY_ENTITY_DOMAINS = [
+    BINARY_SENSOR_DOMAIN,
+    SWITCH_DOMAIN,
+    LIGHT_DOMAIN,
+    INPUT_BOOLEAN_DOMAIN
+]
+
+
+def binary_entity_id(value: str) -> str:
```

Comment:
Can we use `cv.entities_domain` or is that more of a hassle than it saves?

Reply:
I think it makes it more complicated, and I'm not sure if that will work for things that are using the new embedded platform approach.

Reply:
Entities of embedded platforms have the same ID structure as before, so that's not a problem.

Reply:
So we'd effectively be doing something like this?

```
binary_entity_id = vol.All(
    cv.entity_id,
    vol.Any(
        cv.entity_domain(BINARY_SENSOR_DOMAIN),
        cv.entity_domain(SWITCH_DOMAIN),
        cv.entity_domain(LIGHT_DOMAIN),
        cv.entity_domain(INPUT_BOOLEAN_DOMAIN)
    )
)
```

I don't think it changes things much, but makes it so we have to write one less method, so I'm down for that.

Reply:
Good, I agree.